### ISSUE TITLE
[`switch-case-braces`] Add option to allow exceptions for single line case bodies

### ISSUE THREAD

**Issue by RebeccaStevens** [sindresorhus/eslint-plugin-unicorn#1932]
Add an option to treat cases that immediately break or return differently.

```js
function foo(param) {
  switch (param) {
    case bar:
      break;
    case baz:
      return "baz";
    case qux: {
      abc = 123;
      break;
    }
  }
  return "quux";
}
```

**Comment by fisker** [comment]
Is [`'avoid'` option](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/switch-case-braces.md#options) you are looking for?

**Comment by RebeccaStevens** [comment]
No, I want them to be require in the case of multiline blocks, and avoid for single line blocks.

Though, actually I'd probably want a "consistent" option as well. If I use braces for one block, all blocks in that `switch` should have them.

**Comment by fisker** [comment]
```
	['error', 'always', {exception: ['BreakStatement', 'ReturnStatement']}]
```
?

**Comment by RebeccaStevens** [comment]
yeah, that would work.

**Comment by fregante** [comment]
I don't particularly like this rule, but I don't see why those statements should be treated any differently.

**Comment by RebeccaStevens** [comment]
Basically, enforce braces in this case:

```js
switch (value) {
  case "a": {
    perform
    some
    action
    break;
  }
  case "b": {
    perform
    some
    other
    action
    break;
  }
}
```

and disallow them in this case:

```js
switch (value) {
  case "a":
    return 1;
  case "b":
    return 2;
}
```

**Comment by some-user123** [comment]
I was just looking for the same thing and luckily found this issue. For me, the point is that

```js
switch (value) {
    case 'a': {
      return 1
    }
    case 'b': {
      return 2
    }
  }
```

looks ugly and takes more space while not adding any value...

**Comment by fisker** [comment]
How do you think a new option `single-statement`. Enforce the switch case to be a single node, when it's not it should be wrapped in braces to be a single `BlockStatement`.

Inspired by [`useSingleCaseStatement` rule from rome](https://docs.rome.tools/lint/rules/usesinglecasestatement/).
